image/helpers.go

```diff
@@ -404,3 +406,57 @@ func (tsto *ToolingStore) Find(at *asserts.AssertionType, headers map[string]str
 	}
 	return tsto.sto.Assertion(at, pk, tsto.user)
 }
+
+// var so that it can be mocked for tests
+var writeResolvedContent = writeResolvedContentImpl
+
+// writeResolvedContent takes gadget.Info and the unpacked
+// gadget/kernel snaps and outputs the resolved content from the
+// {gadget,kernel}.yaml into a filesystem tree with the structure:
+// <prepareImageDir>/resolved-content/<volume-name>/<structure-name>/...
+//
+// E.g.
+// /tmp/prep-img/resolved-content/pi/ubuntu-seed/{config.txt,bootcode.bin,...}
+func writeResolvedContentImpl(prepareDir string, info *gadget.Info, gadgetUnpackDir, kernelUnpackDir string) error {
+	fullPrepareDir, err := filepath.Abs(prepareDir)
+	if err != nil {
+		return err
+	}
+	targetDir := filepath.Join(fullPrepareDir, "resolved-content")
+
+	constraints := gadget.LayoutConstraints{
+		NonMBRStartOffset: 1 * quantity.OffsetMiB,
```

Comment:
It's pretty annoying that we need to keep repeating this across the codebase. Maybe we should add something like
```
var DefaultLayoutConstraints = LayoutConstraints{
	NonMBRStartOffset: 1 * quantity.OffsetMiB,
}
```
in the gadget package. I'll take a look into this in a separate PR.

Reply:
Great suggestions, I will also do this via a followup

Reply:
Actually I did it in this one now.